### PR TITLE
Fix issue with nil properties not being set and passing hash with string keys issue

### PR DESCRIPTION
**Pull request by ErickFabian**
Fixes:
- Properties being set to nil would not be updated

### CODE REVIEW

**Review by javierg — CHANGES_REQUESTED** [review]
These are 2 issues, can you create 2 different PRs?

**Comment by ErickFabian** [comment]
@javierg  Done, this is the other PR https://github.com/amco/dolly/pull/165

**Comment by javierg** [comment]
@ErickFabian  can you write a test please.  I am not sure if this change will make any missing attribute on the model being removed.

**Comment by ErickFabian** [comment]
@javierg I've added a spec for setting nil to a property on a doc

File: test/property_manager_test.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'test_helper'
+
+class TestDoc < Dolly::Document
+  property :name, class_name: String
+end
+
+class PropertyManagerTest < Test::Unit::TestCase
+  test 'write_attribute with nil value' do
+    doc = TestDoc.new(name: 'name')
```
Comment (javierg) [inline test/property_manager_test.rb]: please add more properties, other than the one being updated

**Review by javierg — APPROVED** [review]
:shipit:

**Pull request merged**